### PR TITLE
Development sprint 1

### PR DESCRIPTION
**Pull request by bmallred**
This is the pull request concerning the `NODE_ENV` environment variable. On TravisCI for the staging repository we just added it to the TravisCI settings (even though it is not sensitive). We can work out a better place for it, but as long as the environment variable is available to the following command it should be okay:

```
docker-compose run frontend npm run build --silent
```

found in [.travis.yml](https://github.com/18F/e-QIP-prototype/blob/master/.travis.yml#L27)

Additional conversation at [PR #86](https://github.com/truetandem/e-QIP-prototype/pull/86)

### CODE REVIEW

**Comment by bmallred** [comment]
This is the pull request concerning the `NODE_ENV` environment variable. On TravisCI for the staging repository we just added it to the TravisCI settings (even though it is not sensitive). We can work out a better place for it, but as long as the environment variable is available to the following command it should be okay:

```
docker-compose run frontend npm run build --silent
```

found in [.travis.yml](https://github.com/18F/e-QIP-prototype/blob/master/.travis.yml#L27)

Additional conversation at [PR #86](https://github.com/truetandem/e-QIP-prototype/pull/86)

**Comment by adelevie** [comment]
> On TravisCI for the staging repository we just added it to the TravisCI settings (even though it is not sensitive).

I'd prefer this get checked in to the repo. The fewer env vars we have to track manually, the better.

File: bin/deploy.sh [hunk 1]
```diff
@@ -4,33 +4,26 @@ API="https://api.fr.cloud.gov"
...
   FRONTEND_NAME="eqip-prototype-staging"
   FRONTEND_MANIFEST="manifest-frontend-staging.yml"
-
-  cf zero-downtime-push $API_NAME -f $API_MANIFEST
-  cf zero-downtime-push $FRONTEND_NAME -f $FRONTEND_MANIFEST
 else
   echo "Unknown space: $SPACE"
   exit
 fi
+
```
Comment (adelevie) [inline bin/deploy.sh]: nice refactor ... did you confirm this works on staging?

File: src/config/environment.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Env {
+  ApiBaseURL () { return 'http://localhost:3000' }
```
Comment (adelevie) [inline src/config/environment.js]: Eventually turn this into an env var/VCAP_SERVICES value.

File: src/config/environment.production.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class Env {
+  ApiBaseURL () { return 'https://eqip-prototype-api.fr.cloud.gov' }
```
Comment (adelevie) [inline src/config/environment.production.js]: Thoughts on turning this into config?

File: src/config/environment.staging.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+class Env {
+  ApiBaseURL () { return 'https://eqip-prototype-api-staging.fr.cloud.gov' }
```
Comment (adelevie) [inline src/config/environment.staging.js]: Same here.

File: src/services/api.js [hunk 5]
```diff
@@ -1,24 +1,25 @@
 import axios from 'axios'
+import { env } from '../config'
 
 class Api {
   constructor () {
     this.proxy = axios.create({
-      baseURL: 'http://localhost:3000',
+      baseURL: env.ApiBaseURL(),
```
Comment (adelevie) [inline src/services/api.js]: Very nice.

File: src/services/api.js [hunk 6]
```diff
@@ -51,38 +52,38 @@ class Api {
...
   }
 
   validateState (state) {
-    return this.proxySecured.get(`/validate/address/${state}`)
+    return this.proxySecured.get(env.EndpointValidateState(state))
   }
 
   validateAddress (address) {
-    return this.proxySecured.post('/validate/address', address)
+    return this.proxySecured.post(env.EndpointValidateAddress(), address)
```
Comment (adelevie) [inline src/services/api.js]: Good use of the `env` module (or class?)

File: src/config/environment.production.js (hunk 3 above)
Comment (bmallred) [inline src/config/environment.production.js]: The Javascript is compiled and not sure we have access to environment variables in a static `.js` file.

Are we running this on a NodeJS instance? May have access to it then.

File: bin/deploy.sh (hunk 1 above)
Comment (bmallred) [inline bin/deploy.sh]: Yeah looks good according to build [230](https://travis-ci.org/truetandem/e-QIP-prototype/builds/180746513)

File: src/services/api.js (hunk 6 above)
Comment (bmallred) [inline src/services/api.js]: `env` is the exported instantiation (ah words) of the `Env` class over at `src/config/environment.js`.

File: .travis.yml [hunk 7]
```diff
@@ -10,6 +10,7 @@ before_install:
   - sudo mv docker-compose /usr/local/bin
 
 script:
+  - sed -i "s/NODE_ENV=/NODE_ENV=$(echo $NODE_ENV)/" .env
```
Comment (adelevie) [inline .travis.yml]: Sorry to be late to this --- what does this command do? I'm not experienced with `sed`.

Comment (bmallred) [inline .travis.yml]: Replaces the line in `NODE_ENV=` in `.env` with whatever the TravisCI environment variable is set to so docker builds the for the right environment.

The `s/` command is substitution... so `s/<what we are looking to replace>/<what we want to replace it with>/`

**Pull request merged**